caas/broker.go

```diff
@@ -170,31 +169,56 @@ type Broker interface {
 	// Provider returns the ContainerEnvironProvider that created this Broker.
 	Provider() ContainerEnvironProvider
 
-	// APIVersion returns the master kubelet API version.
+	// InstancePrechecker provides a means of "prechecking" placement
+	// arguments before recording them in state.
+	environs.InstancePrechecker
+
+	// BootstrapEnviron defines methods for bootstrapping a controller.
+	environs.BootstrapEnviron
+
+	// ResourceAdopter defines methods for adopting resources.
+	environs.ResourceAdopter
+
+	// StorageValidator provides methods to validate storage.
+	StorageValidator
```

Comment:
the method `ValidateStorageClass` in `StorageValidator` is k8s spec as well

Reply:
Can't we repurpose this for validating storage for caas in general? It accepts a map so we should be able to make it work with ecs, right?

Reply:
I mean the `ValidateStorageClass` method name is purely for k8s.
But that's ok we can change it later if we need.